lib/markdown2.py

```diff
@@ -390,26 +390,37 @@ def preprocess(self, text):
     #   another-var: blah blah
     #
     #   # header
-
-    _metadata_pat = re.compile(r"""
-        ^
-        (?:---[\ \t]*\n)?                       # optional "---"
-        ((?:[ \t]*[^ \t:]+[\ \t]*:[^\n]*\n)+)   # "key: value" pairs
-        (?:---[ \t]*)?                          # optional "---"
-        \n""",
-        re.VERBOSE
-    )
+    p = re.compile(r'^(?:---[\ \t]*\n)?(.*:\s+>\n\s+[\S\s]+?)(?=\n\w+\s*:\s*\w+\n|\Z)|([\S\w]+\s*:(?! >)[ \t]*.*\n?)(?:---[\ \t]*\n)?', re.MULTILINE)
```

Comment:
You are not overlooking anything. It was just more convenient for debugging. If you want I can change it.